### ISSUE TITLE
Improve image output in feeds

### ISSUE THREAD

**Issue by tnorthcutt** [hlashbrooke/Seriously-Simple-Podcasting#103]
Currently, if an image is set, [SSP outputs an `image` element with an `href` element](https://github.com/hlashbrooke/Seriously-Simple-Podcasting/blob/c4dcf9a6016df699b27819f8b3acfd75d47e4a97/templates/feed-podcast.php#L269):

`<image href="<?php echo esc_url( $image ); ?>"></image>`

According to the [RSS 2.0 Specification](https://cyber.law.harvard.edu/rss/rss.html):

>
```
<image> is an optional sub-element of <channel>, which contains three required and three optional sub-elements.
<url> is the URL of a GIF, JPEG or PNG image that represents the channel. 
<title> describes the image, it's used in the ALT attribute of the HTML <img> tag when the channel is rendered in HTML. 
<link> is the URL of the site, when the channel is rendered, the image is a link to the site. (Note, in practice the image <title> and <link> should have the same value as the channel's <title> and <link>. 
```

Based on that, we should modify the image element we output. :thought_balloon: 

However, we also [currently](https://github.com/hlashbrooke/Seriously-Simple-Podcasting/blob/c4dcf9a6016df699b27819f8b3acfd75d47e4a97/templates/feed-podcast.php#L297) do `do_action( 'rss2_head' );` which results in core's [`rss2_site_icon()`](https://github.com/WordPress/WordPress/blob/0e53ff669279c585e01d8ac8fbb49e80526b513e/wp-includes/feed.php#L580) function [being called](https://github.com/WordPress/WordPress/blob/0e53ff669279c585e01d8ac8fbb49e80526b513e/wp-includes/default-filters.php#L264), which outputs a properly formatted `image` element, but not with the data from SSP's settings. :cry: 

I propose we add `remove_action( 'rss2_head', 'rss2_site_icon' );` to prevent this.

Let me know if this would be a welcome change, and I'll put together a PR for it.

**In summary:**
This plugin is awesome, and I'd love to help make it more awesomer. :notes:

**Comment by hlashbrooke** [comment]
Hey @tnorthcutt! Sorry for the delayed response here - just haven't had time to look at this in the last couple of weeks. Thanks for looking into this issue so thoroughly - I agree that we need to patch it up, so if you could submit the PR for that, then that would be great :)

I'd say use `remove_action` like you are proposing, then modify `feed-podcast.php` to output a properly formatted element using the SSP settings. Let me know if you're happy to do that, otherwise I'll work on it when I get a moment :)

**Comment by tnorthcutt** [comment]
@hlashbrooke https://github.com/hlashbrooke/Seriously-Simple-Podcasting/pull/104 :+1:

**Comment by hlashbrooke** [comment]
Fixed in #104
